zeroconf/__init__.py

```diff
@@ -2969,24 +2948,23 @@ def send(self, out: DNSOutgoing, addr: Optional[str] = None, port: int = _MDNS_P
     def close(self) -> None:
         """Ends the background threads, and prevent this instance from
         servicing further queries."""
-        if not self._GLOBAL_DONE:
-            # remove service listeners
-            self.remove_all_service_listeners()
-            self.unregister_all_services()
-            self._GLOBAL_DONE = True
-
-            # shutdown recv socket and thread
-            if not self.unicast:
-                self.engine.del_reader(cast(socket.socket, self._listen_socket))
-                cast(socket.socket, self._listen_socket).close()
-            if self.multi_socket:
-                for s in self._respond_sockets:
-                    self.engine.del_reader(s)
-            self.engine.join()
-
-            # shutdown the rest
-            self.notify_all()
-            self.notify_reaper()
-            self.reaper.join()
+        if self._GLOBAL_DONE:
+            return
+        # remove service listeners
+        self.remove_all_service_listeners()
+        self.unregister_all_services()
+        self._GLOBAL_DONE = True
+
+        # shutdown recv socket and thread
+        if not self.unicast:
+            self.engine.del_reader(cast(socket.socket, self._listen_socket))
+            cast(socket.socket, self._listen_socket).close()
+        if self.multi_socket:
             for s in self._respond_sockets:
-                s.close()
+                self.engine.del_reader(s)
+        self.engine.join()
+
+        # shutdown the rest
+        self.notify_all()
+        for s in self._respond_sockets:
+            s.close()
```

Comment:
The actual change in this method is just `self.notify_reaper()` removal, right? The rest is just early return + deindenting the remaining code instead of keeping the code inside the `if` block?

Reply:
Yes removal of `notify_reaper` and the guard

Reply:
I like decreasing the indentation level, just not in the same PR as it makes it more difficult to see what's changing logically and what's just being reformatted. :)

Reply:
Is your preference to revert the guard here and make a new one after this?

Reply:
Sorry that was making it harder to follow.  I moved it to another commit, and then rebased on top.

Reply:
No worries at all, thanks.